django/db/models/query.py

```diff
@@ -53,22 +53,15 @@ async def _async_generator(self):
         # then repeatedly dispatch to it in a sync thread.
         sync_generator = self.__iter__()
 
-        # StopIteration cannot be raised up through sync_to_async() as too much
-        # of the Python async system was originally built on generators.
-        # Instead, capture it inside a custom sync function and change it to an
-        # exception that *can* be propagated.
-        # At some point, this could be moved to asgiref.
-        def custom_next(gen):
-            try:
-                return next(gen)
-            except StopIteration:
-                raise StopAsyncIteration()
+        def next_slice(gen):
+            return list(islice(gen, self.chunk_size))
 
         while True:
-            try:
-                yield await sync_to_async(custom_next)(sync_generator)
-            except StopAsyncIteration:
-                return
+            chunk = await sync_to_async(next_slice)(sync_generator)
+            for item in chunk:
+                yield item
```

Comment:
`yield from` is not allowed in async functions.

Reply:
thanks, TIL